### PR TITLE
Add GitLab CI

### PR DESCRIPTION
**Pull request by LipuFei**

### CODE REVIEW

**Comment by LipuFei** [comment]
@nallath Could you review and merge this?

File: scripts/check_material_profiles_new_with_lxml.py [hunk 1]
```diff
@@ -51,14 +54,15 @@ def _getAllMaterialsContentsInDir(self, directory: str) -> Dict[str, Optional[st
             break
         return result
 
-    def validateAll(self, directory: str) -> bool:
-        materials_dir = self._getMaterialsDir(os.path.abspath(directory))
+    def validateAll(self, repo_root_dir: str) -> bool:
+        materials_dir = self._getMaterialsDir(repo_root_dir)
+        fdmmaterial_xsd_file_path = os.path.join(repo_root_dir, "scripts", "fdmmaterial.xsd")
```
Comment (Ghostkeeper) [inline scripts/check_material_profiles_new_with_lxml.py]: You should still call `os.path.abspath` on the result here, lest the `repo_root_dir` is a relative path.

Comment (LipuFei) [reply scripts/check_material_profiles_new_with_lxml.py]: Fixed in 3796512

**Comment by LipuFei** [comment]
@Ghostkeeper @nallath Could you review it again? I have fixed the merge conflicts and the remarks.

**Pull request merged**